x/oracle/keeper/genesis.go

```diff
@@ -43,8 +43,12 @@ func (k Keeper) IterateAllMedianPrices(
 	for ; iter.Valid(); iter.Next() {
 		var decProto sdk.DecProto
 		k.cdc.MustUnmarshal(iter.Value(), &decProto)
-		denom := types.ParseDenomFromMedianKey(iter.Key())
-		median := types.ExchangeRateTuple{ExchangeRate: decProto.Dec, Denom: denom}
+		denom := types.ParseDenomFromKey(iter.Key(), types.KeyPrefixMedian)
+		blockNum := types.ParseBlockFromKey(iter.Key())
```

Comment:
can we have one function which will return both denom and a block number?

Reply:
Done👍